File: apiwznm/WznmQMtdList.cpp
/**
	* \file WznmQMtdList.cpp
	* API code for table TblWznmQMtdList (implementation)
	* \author Alexander Wirthmueller
	* \date created: 11 Jul 2020
	* \date modified: 11 Jul 2020
	*/

#include "WznmQMtdList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQMtdList
 ******************************************************************************/

WznmQMtdList::WznmQMtdList(
			const uint jnum
			, const string sref
			, const string stubRefWznmMJob
			, const string yesnoExecsrv
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->stubRefWznmMJob = stubRefWznmMJob;
	this->yesnoExecsrv = yesnoExecsrv;
};

bool WznmQMtdList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQMtdList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "stubRefWznmMJob", "job", stubRefWznmMJob);
		extractStringUclc(docctx, basexpath, "yesnoExecsrv", "exs", yesnoExecsrv);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQMtdList
 ******************************************************************************/

ListWznmQMtdList::ListWznmQMtdList() {
};

ListWznmQMtdList::ListWznmQMtdList(
			const ListWznmQMtdList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQMtdList(*(src.nodes[i]));
};

ListWznmQMtdList& ListWznmQMtdList::operator=(
			const ListWznmQMtdList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQMtdList(*(src.nodes[i]));

	return *this;
};

ListWznmQMtdList::~ListWznmQMtdList() {
	clear();
};

void ListWznmQMtdList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQMtdList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQMtdList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQMtdList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQMtdList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQMtdList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQMtdList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwznm/WznmQMtdList.java
/**
	* \file WznmQMtdList.h
	* Java API code for record of table TblWznmQMtdList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQMtdList {
	
	public WznmQMtdList(
				int jnum
				, String sref
				, String stubRefWznmMJob
				, String yesnoExecsrv
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.stubRefWznmMJob = stubRefWznmMJob;
		this.yesnoExecsrv = yesnoExecsrv;
	};

	public int jnum;
	public String sref;
	public String stubRefWznmMJob;
	public String yesnoExecsrv;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQMtdList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			stubRefWznmMJob = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMJob", "job", null, 0);
			yesnoExecsrv = Xmlio.extractStringUclc(doc, basexpath, "yesnoExecsrv", "exs", null, 0);
			
			return true;
		};

		return false;
	};
};

